unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatHUDController.cs

```diff
@@ -0,0 +1,67 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine.Events;
+using ChatMessage = ChatController.ChatMessage;
+using ChatMessageType = ChatController.ChatMessageType;
+public class ChatHUDController : IDisposable
+{
+    public const int MAX_CHAT_ENTRIES = 100;
+
+    public ChatHUDView view;
+
+    public UnityAction<string> OnSendMessage;
+
+    public void Initialize(ChatHUDView view = null, UnityAction<string> onSendMessage = null)
+    {
+        if (view == null)
+        {
+            this.view = ChatHUDView.Create();
+        }
+        else
+        {
+            this.view = view;
+        }
+
+        this.view.Initialize(this, onSendMessage);
+    }
+
+    public void AddChatMessage(ChatMessage message)
+    {
+        view.AddEntry(message);
+
+        if (view.entries.Count > MAX_CHAT_ENTRIES)
+        {
+            var result = TrimAndSortChatMessages(view.entries.Select((x) => x.message).ToList());
+            view.RepopulateAllChatMessages(result);
+        }
+    }
+
+    public void FilterByType(ChatMessageType type)
+    {
+        var result = ChatController.i.entries.Where((x) => x.messageType == type).ToList();
+        result = TrimAndSortChatMessages(result);
+        view.RepopulateAllChatMessages(result);
```

Comment:
Is it necessary to repopulate the entries list consantly? can't we just disable the gameobject of the messages we don't want to show (the ui would adapt automatically) and re-enable them when we need to show them again? for example when filtering messages.
Because right now all this sorting and traversing will happen every time we get a message, after we've reached the MAX_CHAT_ENTRIES right?

the next comment is related, please check that one also

Reply:
Going to give a shot of optimizing it as you suggested.